protocols/virtualdefund/virtualdefund.go

```diff
@@ -121,33 +128,41 @@ func NewObjective(request ObjectiveRequest,
 		return Objective{}, fmt.Errorf("client address not found in an expected participant index")
 
 	}
+
+	if largestPaymentAmount == nil {
+
+		largestPaymentAmount = big.NewInt(0)
+	}
+
+	finalOutcome := outcome.SingleAssetExit{}
+	// Since is Alice is responsible for issuing vouchers she always has the largest payment amount
+	// This means she can just set her FinalOutcomeFromAlice based on the largest voucher amount she has sent
+	if myAddress == alice {
+
+		finalOutcome = initialOutcome.Clone()
+		finalOutcome.Allocations[0].Amount.Sub(finalOutcome.Allocations[0].Amount, largestPaymentAmount)
+		finalOutcome.Allocations[1].Amount.Add(finalOutcome.Allocations[1].Amount, largestPaymentAmount)
+
+	}
+
 	return Objective{
-		Status:         status,
-		InitialOutcome: initialOutcome,
-		PaidToBob:      request.PaidToBob,
-		VFixed:         V.FixedPart,
-		Signatures:     [3]state.Signature{},
-		MyRole:         V.MyIndex,
-		ToMyLeft:       toMyLeft,
-		ToMyRight:      toMyRight,
+		Status:               status,
+		InitialOutcome:       initialOutcome,
+		FinalOutcome:         finalOutcome,
+		MinimumPaymentAmount: largestPaymentAmount,
+		VFixed:               V.FixedPart,
+		Signatures:           [3]state.Signature{},
+		MyRole:               V.MyIndex,
+		ToMyLeft:             toMyLeft,
+		ToMyRight:            toMyRight,
 	}, nil
-
 }
 
 // calculatePaidToBob determines the amount paid to bob by comparing the prefund setup state and the proposed final state.
-func calculatePaidToBob(proposedFinalState state.State, getChannel GetChannelByIdFunction) (*big.Int, error) {
-	if !proposedFinalState.IsFinal {
-		return big.NewInt(0), fmt.Errorf("expected final state")
-	}
-	cId := proposedFinalState.ChannelId()
-	c, found := getChannel(cId)
-	pf := c.PreFundState()
+func calculatePaidToBob(initialOutcome outcome.Exit, finalOutcome outcome.Exit) (*big.Int, error) {
 
-	if !found {
-		return big.NewInt(0), fmt.Errorf("could not find channel %s", cId)
-	}
-	initialBobAmount := pf.Outcome[0].Allocations[1].Amount
-	finalBobAmount := proposedFinalState.Outcome[0].Allocations[1].Amount
+	initialBobAmount := initialOutcome[0].Allocations[1].Amount
+	finalBobAmount := finalOutcome[0].Allocations[1].Amount
 	return big.NewInt(0).Sub(finalBobAmount, initialBobAmount), nil
```

Comment:
Does Bob need to check Alice's slot, too? What if she increased her balance and Bob's balance? Then the channel would be underfunded, and someone other than Alice will end up losing out! I don't think we have an on chain check for this, so we had better have an off chain check!

Reply:
Good point!

Reply:
I've updated this in https://github.com/statechannels/go-nitro/pull/866/commits/bd0ec246ad4631d0d97d5337f6bdfbae3e4f31a6.

Now when receiving the final signed state from Alice we check:
- The Destinations in the outcome are correct
- The amount paid to bob is added to bob's allocation
- The amount paid to bob is subtracted from Alice's allocation
- If we're Bob that the payment amount is greater than the latest voucher we have

Reply:
@geoknee since this is security related it might be worth you taking a look at https://github.com/statechannels/go-nitro/commit/bd0ec246ad4631d0d97d5337f6bdfbae3e4f31a6 and making sure those checks are sufficient/correct.

Reply:
They look good to me, but it would be nice to have some tests that try and break them :-)